### PR TITLE
Add methods to get padded and ragged arrays to FlattenedStorage

### PR DESCRIPTION
**Pull request by pmrv**
I want this for https://github.com/pyiron/pyiron_atomistics/pull/387 and I guess it could also be useful the get neighbors class.

### CODE REVIEW

File: pyiron_base/generic/flattenedstorage.py [hunk 1]
```diff
@@ -339,6 +339,61 @@ def get_array(self, name, frame=None):
         else:
             raise KeyError(f"no array named {name}")
 
+    def get_array_ragged(self, name):
```
Comment (sudarsan-surendralal) [inline pyiron_base/generic/flattenedstorage.py]:
```suggestion
    def get_array_ragged(self, name: str) -> np.ndarray:
```

File: pyiron_base/generic/flattenedstorage.py [hunk 2]
```diff
@@ -339,6 +339,61 @@ def get_array(self, name, frame=None):
...
+
+        Returns:
+            numpy.ndarray, dtype=object: ragged arrray of all elements in all chunks
+        """
+        if name in self._per_chunk_arrays:
+            return self.get_array(name)
+        return np.array([self.get_array(name, i) for i in range(len(self))],
+                        dtype=object)
+
+    def get_array_filled(self, name):
```
Comment (sudarsan-surendralal) [inline pyiron_base/generic/flattenedstorage.py]: Type hinting here too

File: pyiron_base/generic/flattenedstorage.py [hunk 3]
```diff
@@ -339,6 +339,61 @@ def get_array(self, name, frame=None):
...
+        max_len = self._per_chunk_arrays["length"].max()
+        def resize_and_pad(v):
+            l = len(v)
+            per_shape = self._per_element_arrays[name].shape[1:]
+            v = np.resize(v, max_len * np.prod(per_shape, dtype=int))
+            v = v.reshape((max_len,) + per_shape)
+            if name in self._fill_values:
+                fill = self._fill_values[name]
+            else:
+                fill = {np.dtype("int32"):   -1,
```
Comment (sudarsan-surendralal) [inline pyiron_base/generic/flattenedstorage.py]: Just a suggestion: I feel this dict of default fill values could be defined outside this function. Might be easier to extend/modify

**Review by sudarsan-surendralal — APPROVED** [review]
LGTM. Just a few small typing/code style suggestions.

File: pyiron_base/generic/flattenedstorage.py (hunk 2 above)
Comment (pmrv) [reply pyiron_base/generic/flattenedstorage.py]:
```suggestion
    def get_array_filled(self, name: str) -> np.ndarray:
```

**Pull request merged**